lib/ruby_indexer/test/index_test.rb

```diff
@@ -1472,5 +1472,26 @@ class Foo
 
       assert_empty(@index.method_completion_candidates("bar", "Foo"))
     end
+
+    def test_resolving_constants_in_aliased_namespace
+      index(<<~RUBY)
+        module Original
+          module Something
+            CONST = 123
+          end
+        end
+
+        module Other
+          ALIAS = Original::Something
+        end
+
+        module Third
+          Other::ALIAS::CONST
+        end
+      RUBY
+
+      entry = T.must(@index.resolve("Other::ALIAS::CONST", ["Third"])&.first)
+      refute_nil(entry)
```

Comment:
```suggestion
      refute_empty(@index.resolve("Other::ALIAS::CONST", ["Third"]))
```

Reply:
When we fail to resolve a constant, we return `nil`, not empty array. This would fail with `undefined method empty? for nil:NilClass`.

Reply:
So I guess you can `assert_nil(@index.resolve("Other::ALIAS::CONST", ["Third"]))`

Reply:
Actually, this test was really weak. I changed it to assert the kind of entry we get back and that it points to the correct name.